hazelcast/src/test/java/com/hazelcast/client/AuthenticationInformationLeakTest.java

```diff
@@ -87,29 +88,34 @@ public void cleanUp() {
     }
 
     @Test
-    public void testAuthenticationExceptionDoesNotLeakInfo() throws IOException {
+    public void testAuthenticationExceptionDoesNotLeakInfo() {
         SerializationService ss = new DefaultSerializationServiceBuilder().build();
         InetSocketAddress endpoint = instance.getCluster().getLocalMember().getSocketAddress();
-        try (Socket socket = new Socket()) {
-            socket.setReuseAddress(true);
-            socket.connect(endpoint);
-            ClientMessage res;
-            try (OutputStream os = socket.getOutputStream(); InputStream is = socket.getInputStream()) {
-                os.write(CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
-                res = getKeyValue(ss, os, is);
+        // The server sends an exception message and immediately closes the connection without waiting the message to
+        // be written. It's possible that we receive EOF from socket, which will make ClientTestUtil.readResponse throw.
+        // Therefore, we use assertTrueEventually here.
+        assertTrueEventually(() -> {
```

Comment:
The assertTrueEventually still does not solve the problem we have. If the lambda throws any exception other than the assertion exception, it terminates by rethrowing that exception. Here, we will likely to get some sort of IOException from `res = getKeyValue(ss, os, is);` line. And, when we get that, the test will fail like before.

Reply:
Makes sense. Did a refactor. can you check now?